crates/tabby-index/src/structured_doc/mod.rs

```diff
@@ -37,7 +38,7 @@ impl IndexAttributeBuilder<StructuredDoc> for StructuredDocBuilder {
     async fn build_chunk_attributes<'a>(
         &self,
         document: &'a StructuredDoc,
-    ) -> BoxStream<'a, JoinHandle<(Vec<String>, serde_json::Value)>> {
+    ) -> BoxStream<'a, JoinHandle<(Result<Vec<String>>, serde_json::Value)>> {
```

Comment:
```suggestion
    ) -> BoxStream<'a, JoinHandle<(Result<Vec<String>, serde_json::Value)>>> {
```

in future, `serde_json::Value` might contains a value coming from a fail-able source.